README.md

```
@@ -382,13 +434,39 @@ TODO
+   ```
+5. **Commit your changes**: `git commit -am 'Add some feature'`
+6. **Push to the branch**: `git push origin my-new-feature`
+7. **Submit a pull request**!
+
+### Development Tools
+- **Poetry**: Dependency management
+- **ruff**: Ultra-fast linting and formatting
+- **pytest**: Testing framework with 127 passing tests
+- **Documentation**: See [CLAUDE.md](CLAUDE.md) for architecture details
+
+### Test Status
```

markdownlint-cli2 (0.17.2)

Heading style
Expected: setext; Actual: atx

(MD003, heading-style)